File: packages/js-sdk/src/regex/detectors/at/svnr.ts
// Austrian Social Insurance Number (SVNR) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { atSvnrChecksum } from '../../validators'

class AtSvnrDetector extends RegexDetector {
  entityType = EntityType.AT_SVNR
  score = 0.85
  contextKeywords = [
    'svnr',
    'sozialversicherungsnummer',
    'versicherungsnummer',
    'social insurance',
  ]
  contextRequired = true
  contextWindow = 50

  pattern = /\b\d{4}[- ]?\d{6}\b/g
  validator = atSvnrChecksum
}

registerRegion('at', AtSvnrDetector)

export { AtSvnrDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/at/svnr.py
"""Austrian Social Insurance Number (SVNR) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import at_svnr_checksum


@register_region("at")
class SvnrDetector(RegexDetector):
    entity_type = EntityType.AT_SVNR
    score = 0.85
    context_keywords = [
        "svnr", "sozialversicherungsnummer", "versicherungsnummer",
        "social insurance",
    ]
    context_required = True
    context_window = 50

    pattern = re.compile(
        r"\b\d{4}[- ]?\d{6}\b"
    )
    validator = staticmethod(at_svnr_checksum)
